src/Database/Query.php

```diff
@@ -211,10 +211,10 @@ public function connection($connection = null)
     public function execute()
     {
         $statement = $this->_connection->run($this);
-        $driver = $this->_connection->driver();
         $typeMap = $this->getSelectTypeMap();
 
         if ($typeMap->toArray() && $this->_typeCastAttached === false) {
+            $driver = $this->_connection->getDriver();
```

Comment:
Why? This is not protected internal access, this is public access from the outside.